integration-tests/api/__tests__/store/cart.js

```diff
@@ -145,6 +145,61 @@ describe("/store/carts", () => {
       }
     })
 
+    it("fails to apply expired discount", async () => {
+      const api = useApi()
+
+      try {
+        await api.post("/store/carts/test-cart", {
+          discounts: [{ code: "EXP_DISC" }],
+        })
+      } catch (error) {
+        expect(error.response.status).toEqual(400)
+        expect(error.response.data.message).toEqual("Discount is expired")
+      }
+    })
+
+    it("fails on discount before start day", async () => {
+      const api = useApi()
+
+      try {
+        await api.post("/store/carts/test-cart", {
+          discounts: [{ code: "PREM_DISC" }],
+        })
+      } catch (error) {
+        console.log(error)
+        expect(error.response.status).toEqual(400)
```

Comment:
If line 165 doesn't fail then this will never be tested. 

Instead do:
```
await expect(api.post("/store/carts/test-cart", {
  discounts: [{ code: "PREM_DISC" }],
})).rejects.toThrow(...)
```